### PR TITLE
Grammarly corrections

### PR DESCRIPTION
**Pull request by u01jmg3**
- Grammarly corrections
- Use "number" rather than "amount" when discussing items that can be counted 🧮
  - [The difference between "number" and "amount"](http://crosstalk.cell.com/blog/the-difference-between-number-and-amount#:~:text=In%20other%20words%2C%20number%20refers,measure%20but%20not%20individually%20count.)

### CODE REVIEW

File: src/Illuminate/Database/Connection.php [hunk 1]
```diff
@@ -333,7 +333,7 @@ public function select($query, $bindings = [], $useReadPdo = true)
 
             // For select statements, we'll simply execute the query and return an array
             // of the database result set. Each element in the array will be a single
-            // row from the database table, and will either be an array or objects.
+            // row from the database table, and will either be an array or object.
```
Comment (u01jmg3) [inline src/Illuminate/Database/Connection.php:336]: Note the docblock only mentions returning an array

File: src/Illuminate/Auth/DatabaseUserProvider.php [hunk 2]
```diff
@@ -122,7 +122,7 @@ public function retrieveByCredentials(array $credentials)
             }
         }
 
-        // Now we are ready to execute the query to see if we have an user matching
+        // Now we are ready to execute the query to see if we have a user matching
```
Comment (X-Coder264) [inline src/Illuminate/Auth/DatabaseUserProvider.php:125]: I'm not a native English speaker but this change looks wrong to me.

https://www.grammarly.com/blog/indefinite-articles-a-and-an/

> Use **a** when the indefinite article comes before a word beginning with a consonant sound
 Use **an** when the indefinite article comes before a word beginning with a vowel sound

Comment (u01jmg3) [reply src/Illuminate/Auth/DatabaseUserProvider.php:125]: Unfortunately the English language can be complicated with various exceptions to the rule.

[This article](https://www.quora.com/Is-it-an-user-or-a-user) has several explanations but I think this one sums things up well.

> _The use of the indefinite articles in English: "a" and "an"—which one of them to use—depends on the first sound of the word that will follow the article. For example, the word "user" is pronounced "yoozer" (the initial "y" sound acts as a consonant) so it must be preceded by an "a," not an "an." Now if the word were "usher," which does not begin with a consonantal sound, the correct article to use would be "an."_

**Comment by taylorotwell** [comment]
It throws off my cascading comments 😭

**Pull request closed without merging**